### PR TITLE
Document for tiup-cluster local config directory

### PR DESCRIPTION
**Pull request by lucklove**
Document for https://github.com/pingcap/tiup/pull/712

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: tiup/tiup-cluster.md [hunk 1]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
 
 该操作会将配置发送到目标机器，重启集群，使配置生效。
 
+> **注意：**
+>
+> `tiup cluster edit-config` 命令不支持直接配置监控，但是可以传递一个本地路径进去间接配置，例如：
```
Comment (lonng) [inline tiup/tiup-cluster.md:371]:
```suggestion
> 对于监控组件，可以通过 `tiup cluster edit-config` 在对应实例上添加自定义配置路径来进行配置自定义，例如：
```

File: tiup/tiup-cluster.md [hunk 2]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+
+```yaml
+---
+
+grafana_servers:
+  - host: 172.16.5.134
+    dashboard_dir: /local/dashboards/dir
```
Comment (lonng) [inline tiup/tiup-cluster.md:378]:
```suggestion
    dashboard_dir: /path/to/local/dashboards/dir
```

File: tiup/tiup-cluster.md [hunk 3]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+
+monitoring_servers:
+  - host: 172.16.5.134
+    rule_dir: /local/rules/dir
```
Comment (lonng) [inline tiup/tiup-cluster.md:382]:
```suggestion
    rule_dir: /path/to/local/rules/dir
```

File: tiup/tiup-cluster.md [hunk 4]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+
+alertmanager_servers:
+  - host: 172.16.5.134
+    config_file: /local/alertmanager.yml
```
Comment (lonng) [inline tiup/tiup-cluster.md:386]:
```suggestion
    config_file: /path/to/local/alertmanager.yml
```

File: tiup/tiup-cluster.md [hunk 5]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+```
+
+路径内容格式如下：
+
+- grafana_servers 的 `dashboard_dir` 字段指定的文件夹中应当含有完整的 `*.json`
+- monitoring_servers 的 `rule_dir` 字段定义的文件夹中应当含有完整的 `*.rules.yml`
+- alertmanager_servers 的 `config_file` 格式请参考[此处](https://github.com/pingcap/tiup/blob/master/templates/config/alertmanager.yml)
+
+这样在 reload 的时候 TiUP 会将中控机上对应的配置上传到目标机器的对应配置目录中，上传之前会删除目标机器中的老的配置文件，所以如果想要修改某一个配置文件，请确保将所有的（包含未修改的）配置文件都放在同一个目录中，例如，要修改 grafana 的 tidb.json，那么需要将 grafana 的 dashboards 目录中所有的 *.json 文件拷贝到本地目录中，再修改 tidb.json，否则最终的目标机器上将缺失其他的 json 文件。
```
Comment (lonng) [inline tiup/tiup-cluster.md:395]:
```suggestion
在执行 `tiup reload` 时，TiUP 会将中控机上对应的配置上传到目标机器的对应配置目录中，上传之前会删除目标机器中的老的配置文件，如果想要修改某一个配置文件，请确保将所有的（包含未修改的）配置文件都放在同一个目录中，例如，要修改 grafana 的 tidb.json，可以先将 grafana 的 dashboards 目录中所有的 *.json 文件拷贝到本地目录中，再修改 tidb.json，否则最终的目标机器上将缺失其他的 json 文件。
```

**Review by lonng — COMMENTED** [review]
LGTM

**Comment by lonng** [comment]
@TomShawn PTAL

File: tiup/tiup-cluster.md [hunk 6]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
 
 该操作会将配置发送到目标机器，重启集群，使配置生效。
 
+> **注意：**
+>
+> 对于监控组件，可以通过 `tiup cluster edit-config` 在对应实例上添加自定义配置路径来进行配置自定义，例如：
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:371]:
```suggestion
> 对于监控组件，可以通过执行 `tiup cluster edit-config` 命令在对应实例上添加自定义配置路径来进行配置自定义，例如：
```

File: tiup/tiup-cluster.md [hunk 7]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+    rule_dir: /path/to/local/rules/dir
+
+alertmanager_servers:
+  - host: 172.16.5.134
+    config_file: /path/to/local/alertmanager.yml
+```
+
+路径内容格式如下：
+
+- grafana_servers 的 `dashboard_dir` 字段指定的文件夹中应当含有完整的 `*.json`
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:391]:
```suggestion
- `grafana_servers` 的 `dashboard_dir` 字段指定的文件夹中应当含有完整的 `*.json`
```

File: tiup/tiup-cluster.md [hunk 8]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+- monitoring_servers 的 `rule_dir` 字段定义的文件夹中应当含有完整的 `*.rules.yml`
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:392]:
```suggestion
- `monitoring_servers` 的 `rule_dir` 字段定义的文件夹中应当含有完整的 `*.rules.yml`
```

File: tiup/tiup-cluster.md [hunk 9]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+- alertmanager_servers 的 `config_file` 格式请参考[此处](https://github.com/pingcap/tiup/blob/master/templates/config/alertmanager.yml)
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:393]:
```suggestion
- `alertmanager_servers` 的 `config_file` 格式请参考[此处](https://github.com/pingcap/tiup/blob/master/templates/config/alertmanager.yml)
```

File: tiup/tiup-cluster.md [hunk 10]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+
+在执行 `tiup reload` 时，TiUP 会将中控机上对应的配置上传到目标机器的对应配置目录中，上传之前会删除目标机器中的老的配置文件，如果想要修改某一个配置文件，请确保将所有的（包含未修改的）配置文件都放在同一个目录中，例如，要修改 grafana 的 tidb.json，可以先将 grafana 的 dashboards 目录中所有的 *.json 文件拷贝到本地目录中，再修改 tidb.json，否则最终的目标机器上将缺失其他的 json 文件。
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:395]:
```suggestion
在执行 `tiup reload` 时，TiUP 会将中控机上对应的配置上传到目标机器对应的配置目录中，上传之前会删除目标机器中已有的旧配置文件。如果想要修改某一个配置文件，请确保将所有的（包含未修改的）配置文件都放在同一个目录中。例如，要修改 Grafana 的 `tidb.json` 文件，可以先将 Grafana 的 `dashboards` 目录中所有的 `*.json` 文件拷贝到本地目录中，再修改 `tidb.json` 文件。否则最终的目标机器上将缺失其他的 JSON 文件。
```

File: tiup/tiup-cluster.md [hunk 11]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+
+> **注意：**
+>
+> 如果配置了 `grafana_servers` 的 `dashboard_dir` 字段，在使用 `tiup cluster rename` 功能时需要注意：
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:399]:
```suggestion
> 如果配置了 `grafana_servers` 的 `dashboard_dir` 字段，在执行 `tiup cluster rename` 命令进行集群重命名时需要注意以下几点：
```

File: tiup/tiup-cluster.md [hunk 12]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+>
+> 1. 本地目录中的 dashboards 包含了集群名信息，需要修改本地 dashboards 中的集群名为新的集群名
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:401]: 这里的 1.2.3 是有先后顺序的操作吗？
```suggestion
> 1. 本地目录中的 `dashboards` 包含了集群名信息，需要修改本地 `dashboards` 中的集群名为新的集群名。
```

File: tiup/tiup-cluster.md [hunk 13]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+> 2. 本地目录中的 dashboards 中的 datasource 也要更新为新的集群名（datasource 是以集群名命名的）
+> 3. 执行 `tiup cluster reload -R grafana`
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:403]:
```suggestion
> 2. 本地目录中 `dashboards` 中的 `datasource` 也要更新为新的集群名（`datasource` 是以集群名命名的）。
> 3. 执行 `tiup cluster reload -R grafana` 命令。
```

File: tiup/tiup-cluster.md (hunk 12 above)
Comment (lucklove) [reply tiup/tiup-cluster.md:401]: 是的，有顺序关系，1和2可以互换，但是3一定要最后一步

File: tiup/tiup-cluster.md [hunk 14]
```diff
@@ -366,6 +366,42 @@ tiup cluster reload prod-cluster
...
+
+- `grafana_servers` 的 `dashboard_dir` 字段指定的文件夹中应当含有完整的 `*.json`
+- `monitoring_servers` 的 `rule_dir` 字段定义的文件夹中应当含有完整的 `*.rules.yml`
+- `alertmanager_servers` 的 `config_file` 格式请参考[此处](https://github.com/pingcap/tiup/blob/master/templates/config/alertmanager.yml)
+
+在执行 `tiup reload` 时，TiUP 会将中控机上对应的配置上传到目标机器对应的配置目录中，上传之前会删除目标机器中已有的旧配置文件。如果想要修改某一个配置文件，请确保将所有的（包含未修改的）配置文件都放在同一个目录中。例如，要修改 Grafana 的 `tidb.json` 文件，可以先将 Grafana 的 `dashboards` 目录中所有的 `*.json` 文件拷贝到本地目录中，再修改 `tidb.json` 文件。否则最终的目标机器上将缺失其他的 JSON 文件。
+
+> **注意：**
+>
+> 如果配置了 `grafana_servers` 的 `dashboard_dir` 字段，在执行 `tiup cluster rename` 命令进行集群重命名时需要注意以下几点：
```
Comment (TomShawn) [inline tiup/tiup-cluster.md:399]:
```suggestion
> 如果配置了 `grafana_servers` 的 `dashboard_dir` 字段，在执行 `tiup cluster rename` 命令进行集群重命名时需要完成以下操作：
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by lilin90** [comment]
Hi @CharLotteiu, would you please help translate the content in this PR and create a corresponding PR to the same file of the same branch to the [pingcap/docs](https://github.com/pingcap/docs) repo?

**Comment by CharLotteiu** [comment]
> Hi @CharLotteiu, would you please help translate the content in this PR and create a corresponding PR for the same file of the same branch to the [pingcap/docs](https://github.com/pingcap/docs) repository?

Hi @lilin90, got it! I will send the PR to the docs repo first thing in the morning. I will request your review when I am done with the PR.

**Comment by lilin90** [comment]
@CharLotteiu Thanks a lot! Take your time.

**Pull request merged**